src/server/pps/cmds/cmds.go

```diff
@@ -380,3 +371,39 @@ All jobs created by a pipeline will create commits in the pipeline's repo.
 	result = append(result, runPipeline)
 	return result, nil
 }
+
+func replaceStrategyAliases(pipelineReader io.Reader) string {
+	// We want to allow for a syntactic suger where the user
+	// can specify a strategy with a string such as "map" or "reduce".
+	// To that end, we check for the "strategy" field and replace
+	// the string with an actual strategy object before we unmarshal
+	// the json spec into a protobuf message
+
+	pipeline, err := gabs.ParseJSONBuffer(pipelineReader)
+	if err != nil {
+		if err == io.EOF {
+			os.Exit(0)
+		} else {
+			pkgcmd.ErrorAndExit("Error parsing JSON: %s", err.Error())
+		}
+	}
+
+	inputs := pipeline.S("inputs")
+	children, err := inputs.Children()
+	if err != nil {
+		pkgcmd.ErrorAndExit("Error parsing spec: inputs is not an array", err.Error())
+	}
+	for _, input := range children {
+		strategyAlias, ok := input.S("strategy").Data().(string)
+		if ok {
+			strat, ok := pach.StrategyAliasMap[strategyAlias]
+			if ok {
+				input.Set(strat, "strategy")
+			} else {
+				pkgcmd.ErrorAndExit("Unrecognized strategy: %s", strategyAlias)
+			}
+		}
+	}
+
+	return pipeline.String()
+}
```

Comment:
Doing this in `pachctl` makes me a little uneasy since it means you can't use the aliases if you're using the API directly. Does `oneof` work for this in the protos?